### PR TITLE
Envtest

### PR DESCRIPTION
**Pull request by LimKianAn**
Add integration tests by using *controller-runtime* pkg *envtest*.

### CODE REVIEW

File: config/samples/postgres.yaml [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
-kind: Namespace
-metadata:
-  name: database
----
 apiVersion: database.fits.cloud/v1
 kind: Postgres
 metadata:
-  namespace: database
-  name: sample-name-a
+  namespace: default
```
Comment (LimKianAn) [inline config/samples/postgres.yaml:4]: Change to prod ns: metal-extension-cloud

File: controllers/suite_test.go [hunk 2]
```diff
@@ -7,28 +7,54 @@
...
+	svcClusterTestEnv *envtest.Environment
+
+	externalYAMLDir     = filepath.Join("..", "external")
+	externalYAMLDirTest = filepath.Join(externalYAMLDir, "test")
+
+	instance = &pg.Postgres{}
+)
+
+func init() {
+}
```
Comment (LimKianAn) [inline controllers/suite_test.go:58]: Remove

File: controllers/postgres_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+		interval = time.Second * 2
+		timeout  = time.Second * 30
+	)
+
+	BeforeEach(func() {})
+	AfterEach(func() {})
+
+	Context("complete postgres instance created", func() {
+		It("should add finalizer to the instance", func() {
+			Eventually(func() bool {
```
Comment (LimKianAn) [inline controllers/postgres_controller_test.go:34]: Add comments that the instance is already created

File: external/test/crd-clusterwidenetworkpolicy.yaml [hunk 4]
```diff
@@ -0,0 +1,192 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (LimKianAn) [inline external/test/crd-clusterwidenetworkpolicy.yaml:1]: Use `helm template charts/postgreslet --show-only templates/configmap-sidecars.yaml` to generate the file

Comment (LimKianAn) [reply external/test/crd-clusterwidenetworkpolicy.yaml:1]: For cm-sidecar.yaml

File: pkg/operatormanager/operatormanager.go [hunk 5]
```diff
@@ -664,43 +664,43 @@ func (m *OperatorManager) toObjectKey(obj runtime.Object, namespace string) (cli
 }
 
 // waitTillOperatorReady waits till the operator pod is ready or timeout is reached, polling the status every period
-func (m *OperatorManager) waitTillOperatorReady(ctx context.Context, timeout time.Duration, period time.Duration) error {
```
Comment (LimKianAn) [inline pkg/operatormanager/operatormanager.go:667]: Delete this func.

File: external/test/crd-clusterwidenetworkpolicy.yaml (hunk 4 above)
Comment (eberlep) [reply external/test/crd-clusterwidenetworkpolicy.yaml:1]: You can also add a name:
`helm template envtest charts/postgreslet --show-only templates/configmap-sidecars.yaml`

File: controllers/suite_test.go [hunk 6]
```diff
@@ -176,30 +173,35 @@ func createConfigMapSidecarConfig() {
 	nsObj.Name = "postgreslet-system"
 	Expect(svcClusterClient.Create(newCtx(), nsObj)).Should(Succeed())
 
-	bytes, err := os.ReadFile(filepath.Join(externalYAMLDirTest, "cm-sidecar.yaml"))
+	bytes, err := os.ReadFile(filepath.Join(externalYAMLDirTest, "configmap-sidecars.yaml"))
 	Expect(err).ToNot(HaveOccurred())
 
 	cm := &core.ConfigMap{}
 	Expect(yaml.Unmarshal(bytes, cm)).Should(Succeed())
+	// todo: Should not be hard-coded
```
Comment (eberlep) [inline controllers/suite_test.go:181]: as discussed, please update the template and inject the correct values while generating the file in the `Makefile`

**Review by eberlep — APPROVED** [review]

**Pull request merged**